skimage/transform/tests/test_thin_plate_splines_transform.py

```diff
@@ -0,0 +1,164 @@
+
+        # Test that _estimated is initialized to False
+        assert tform._estimated is False
+        assert tform.coefficients is None
+        assert tform.src is None
+
+    def test_tps_transform_inverse(self):
+        tps = TpsTransform()
+        with pytest.raises(NotImplementedError):
+            tps.inverse()
+
+    def test_tps_transform_ensure_2d(self):
+        assert_array_equal(_ensure_2d(SRC), SRC)
+        assert_array_equal(_ensure_2d(DST), DST)
+
+        array_1d = np.array([0, 5, 10])
+        expected = np.array([[0], [5], [10]])
+        assert_array_equal(_ensure_2d(array_1d), expected)
+
+        empty_array = np.array([])
+        with pytest.raises(ValueError, match="Array of points can not be empty."):
+            _ensure_2d(empty_array)
+
+        scalar = 5
+        with pytest.raises(ValueError, match="Array must be be 2D."):
+            _ensure_2d(scalar)
+
+        array_3d = np.array([[[0, 5], [10, 15]], [[20, 25], [30, 35]]])
+        with pytest.raises(ValueError, match="Array must be be 2D."):
+            _ensure_2d(array_3d)
+
+        control_pts_less_than_3 = np.array([[0, 0], [0, 0]])
+        with pytest.raises(ValueError, match="Array points less than 3 is undefined."):
+            _ensure_2d(control_pts_less_than_3)
+
+    def test_tps_transform_estimation(self):
+        src = np.array([[0, 1], [-1, 0], [0, -1], [1, 0]])
+        dst = np.array([[0, 0.75], [-1, 0.25], [0, -1.25], [1, 0.25]], dtype=np.float32)
+        tform = TpsTransform()
+
+        # Ensure that the initial state is as expected
+        assert tform.coefficients is None
+        assert tform.src is None
+
+        # Perform estimation
+        tform.estimate(src, dst)
+        assert len(tform.src) > 0
+
+        assert len(tform.coefficients) > 0
+        assert tform.coefficients.shape[0] == src.shape[0] + 3
+
+        np.testing.assert_array_equal(tform.src, src)
+        np.testing.assert_allclose(
+            tform.coefficients[:, 0],
+            np.array([0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0]),
+            rtol=0.1,
+            atol=1e-16,
+        )
+        np.testing.assert_allclose(
+            tform.coefficients[:, 1],
+            np.array([-0.0902, 0.0902, -0.0902, 0.0902, 0.0, 0.0, 1.0]),
+            rtol=0.1,
+            atol=1e-16,
+        )
+
+    def test_tps_transform_estimation_failure(self):
+        # Test the estimate method when the estimation fails
+        tform = TpsTransform()
+        src = np.array([[0, 0], [0, 5], [5, 5], [5, 0]])
+        dst = np.array([[5, 0], [0, 0], [0, 5]])
+
+        # Ensure that the initial state is as expected
+        assert tform._estimated is False
+        assert tform.coefficients is None
+        assert tform.src is None
+
+        # Perform the estimation, which should fail due to the mismatched number of points
+        with pytest.raises(ValueError, match=".*shape must be identical"):
+            tform.estimate(src, dst)
```

Comment:
You could also add tests here if `src` and `dst` have fewer than 3 points, are not 2-dimensional, ...